### PR TITLE
Switch to ia cli to upload dumps

### LINKED ISSUE #5930: Transfer monthly dumps using IA client
I am always frustrated when ol-dumps succeed but the transfer to https://archive.org/details/ol_exports?sort=-publicdate fails.

### TODO: Switch to ia client tool. This will only work in production 'til then
From `scripts/oldump.sh` in #5909
```bash
function archive_dumps() {
    # Copy data dumps to https://archive.org/details/ol_exports?sort=-publicdate
    # For progress on transfers, see:
    # https://catalogd.archive.org/catalog.php?checked=1&all=1&banner=rsync%20timeout
    # TODO: Switch to ia client tool. This will only work in production 'til then
    python /olsystem/bin/uploaditem.py $dump --nowait --uploader=openlibrary@archive.org
    python /olsystem/bin/uploaditem.py $cdump --nowait --uploader=openlibrary@archive.org
}

# Only archive if that caller has requested it and we are not testing.
if [ "$archive" == "--archive" ]; then
    if [[ -z $OLDUMP_TESTING ]]; then
        archive_dumps
    fi
fi
```

### Describe the problem that you'd like solved

Replace the calls to [`/olsystem/bin/uploaditem.py`](https://github.com/internetarchive/olsystem/blob/master/bin/uploaditem.py) above with https://github.com/jjjake/internetarchive

### Additional context

Need to generate `_file.xml` and `_meta.xml` as in:
* https://archive.org/download/ol_cdump_2021-10-13
* https://archive.org/download/ol_dump_2021-10-13

### PR DESCRIPTION
**Pull request by jimman2003**
Fixes #5930

https://archive.org/services/docs/api/internetarchive/cli.html#upload

feature
* [x] Use `ia` v2.2.0 the right version to use
* [x] What must be done to ___configure___ ai? @cclauss `ia --config-file /x.y.z` -- Should we use /olsystem?
* [x] What is the right ___upload___ command for `dump`?
* [x] What is the right ___upload___ command for `cdump`?
* ~Should we create `*_files.xml` with code from `uploaditem.py` or with `ia`?~ Auto-created on IA
* ~Should we create `*_meta.xml` with code from `uploaditem.py` or with `ia`?~ Auto-created on IA
* [ ] Currently `uploaditem.py` is run with a `--no-wait` option.  Does `ia` have a async command? No
    * Should we use parallels? 
* [ ] Update https://github.com/internetarchive/openlibrary/wiki/Generating-Data-Dumps

### Technical

I added only the basic metadata
### Testing

verify that the cli does upload the correct files with metadata
### Stakeholders

@cclauss

### CODE REVIEW

**Comment by cclauss** [comment]
@jjjake Would you be willing to mentor @jimman2003 and I thru this process of upgrading the transfer of our monthly Open Library dumps.

**Comment by jjjake** [comment]
Absolutely @cclauss! I'll take a closer look on Monday.

File: scripts/oldump.sh [hunk 1]
```diff
@@ -102,8 +102,9 @@ ls -lhR
 function archive_dumps() {
     # copy stuff to archive.org
     # TODO: Switch to ia client tool. This will only work in production 'til then
-    python /olsystem/bin/uploaditem.py $dump --nowait --uploader=openlibrary@archive.org
-    python /olsystem/bin/uploaditem.py $cdump --nowait --uploader=openlibrary@archive.org
+     ia $dump $dump.txt.gz --metadata "uploader:openlibrary@archive.org" --metadata "title:$dump" --metadata "collection:ol_exports"
+     ia $cdump $cdump.txt.gz  --metadata "uploader:openlibrary@archive.org" --metadata "title:$cdump" --metadata "collection:ol_exports"
+
```
Comment (cclauss) [inline scripts/oldump.sh:107]:
```suggestion
     log "ia version is v$(ia --version)"  # ia version is v1.9.0
     # Need to pre-generate `${dump}_files.xml` and `${dump}_meta.xml` via a subset of `uploaditeems.py`
     ia upload $dump $dump --metadata "uploader:openlibrary@archive.org" --metadata "title:$dump" --metadata "collection:ol_exports"
     ia upload $cdump $cdump --metadata "uploader:openlibrary@archive.org" --metadata "title:$cdump" --metadata "collection:ol_exports"
```

Comment (jimman2003) [reply scripts/oldump.sh:107]: I think the xml is generated though the metadata argument in the ia tool..will add the log call

Comment (cclauss) [reply scripts/oldump.sh:107]: `$dump` and `$cdump` are identifiers and _directory_ names, not filenames.  We want to transfer all the files in those directories.

File: scripts/oldump.sh [hunk 2]
```diff
@@ -103,8 +103,8 @@ function archive_dumps() {
     # copy stuff to archive.org
     # TODO: Switch to ia client tool. This will only work in production 'til then
     log "ia version is v$(ia --version)"  # ia version is v1.9.0
```
Comment (cclauss) [inline scripts/oldump.sh:105]: From https://pypi.org/project/internetarchive/
```suggestion
    log "ia version is v$(ia --version)"  # ia version is v2.2.0
```

File: scripts/oldump.sh [hunk 3]
```diff
@@ -103,8 +103,8 @@ function archive_dumps() {
...
-    ia upload $dump $dump --metadata "collection:ol_exports" #this needs configuration in home folder
-    ia upload $cdump $cdump --metadata "collection:ol_exports" #this needs configuration in home folder
+    ia upload $dump $dump/ --metadata "collection:ol_exports" #this needs configuration in home folder
+    ia upload $cdump $cdump/ --metadata "collection:ol_exports" #this needs configuration in home folder
```
Comment (cclauss) [inline scripts/oldump.sh:107]:
```suggestion
    ia --config-file olsystem/etc/ia.ini upload $dump  $dump/  --metadata "collection:ol_exports" --retries 300
    ia --config-file olsystem/etc/ia.ini upload $cdump $cdump/ --metadata "collection:ol_exports" --retries 300
```

**Comment by cclauss** [comment]
The job runs in this Docker container:
https://github.com/internetarchive/openlibrary/blob/master/docker-compose.production.yml#L90-L105

**Comment by cclauss** [comment]
@mekarpeles Is it OK to use https://github.com/internetarchive/olsystem/blob/master/etc/ia.ini as our config for the upload of dumps?

**Comment by cclauss** [comment]
Please add `year:${date:0:4}` to the metadata so it will appear on the left side of
https://archive.org/details/ol_exports?sort=-publicdate

File: scripts/oldump.sh [hunk 4]
```diff
@@ -102,8 +102,9 @@ ls -lhR
 function archive_dumps() {
     # copy stuff to archive.org
     # TODO: Switch to ia client tool. This will only work in production 'til then
-    python /olsystem/bin/uploaditem.py $dump --nowait --uploader=openlibrary@archive.org
-    python /olsystem/bin/uploaditem.py $cdump --nowait --uploader=openlibrary@archive.org
+    log "ia version is v$(ia --version)"  # ia version is v1.9.0
+    ia upload $dump $dump/ --metadata "collection:ol_exports" --metadata "year:${date:0:4}"#this needs configuration in home folder
+    ia upload $cdump $cdump/ --metadata "collection:ol_exports" --metadata "year:${date:0:4}"#this needs configuration in home folder
```
Comment (cclauss) [inline scripts/oldump.sh:107]:
```suggestion
    log "ia version is v$(ia --version)"  # ia version is v2.2.0
    ia --config-file=/olsystem/etc/ia.ini upload $dump  $dump/  --metadata "collection:ol_exports" --metadata "year:${date:0:4}"
    ia --config-file=/olsystem/etc/ia.ini upload $cdump $cdump/ --metadata "collection:ol_exports" --metadata "year:${date:0:4}"
```

File: scripts/oldump.sh [hunk 5]
```diff
@@ -102,8 +102,9 @@ ls -lhR
 function archive_dumps() {
     # copy stuff to archive.org
     # TODO: Switch to ia client tool. This will only work in production 'til then
-    python /olsystem/bin/uploaditem.py $dump --nowait --uploader=openlibrary@archive.org
-    python /olsystem/bin/uploaditem.py $cdump --nowait --uploader=openlibrary@archive.org
+    log "ia version is v$(ia --version)"  # ia version is v2.2.0
+    ia --config-file=/olsystem/etc/ia.ini upload $dump  $dump/  --metadata "collection:ol_exports" --metadata "year:${date:0:4}" --retries 300
```
Comment (cclauss) [inline scripts/oldump.sh:106]: Let's set the `mediatype: data` (and `format`?)

Keep in mind that `uploader` is `openlibrary@archive.org` for production.

Comment (jimman2003) [reply scripts/oldump.sh:106]: I think that mediatype:data is the default, no need to set it.

Comment (jimman2003) [reply scripts/oldump.sh:106]: What should be the value of format?

Comment (cclauss) [reply scripts/oldump.sh:106]: https://ia801402.us.archive.org/22/items/ol_cdump_2021-11-30/ol_cdump_2021-11-30_meta.xml

Comment (jimman2003) [reply scripts/oldump.sh:106]: format seems like a not standarded metadata field, so do we need to set it?

Comment (jimman2003) [reply scripts/oldump.sh:106]: will do it for consistancy..

**Comment by cclauss** [comment]
Works on sample payloads.  Will test in full on the 01/01/2022 payload.

**Pull request merged**